### PR TITLE
Added Azure Cloud CNC support to the mso_schema_site_service_graph module

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added Azure Cloud CNC support to the mso_schema_site_service_graph module and fixed mso_schema_template_service_graph module.

### CODE REVIEW

**Review by samiib — CHANGES_REQUESTED** [review]
There are some long lines sanity errors.

File: tests/integration/targets/mso_schema_site_service_graph/tasks/l4_l7_devices.yml [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+              - cloudRsCtxProfileToRegion:
+                  attributes:
+                    tDn: "uni/clouddomp/provp-azure/region-westus2"
+              - cloudCidr:
+                  attributes:
+                    addr: 1.1.1.2/32
+                    primary: "yes"
+
+    - name: Add the subnet to the sab_azure_ctxprof
+      cisco.aci.aci_rest:
```
[the remark this answers is not in the document]
Comment (sajagana) [reply tests/integration/targets/mso_schema_site_service_graph/tasks/l4_l7_devices.yml:94]: aci_cloud_subnet - has some issues, so using aci_rest to manage it.

**Review by samiib — APPROVED** [review]
LGTM! Yes will try and fix the aci_cloud_subnet shortly. :)

File: plugins/module_utils/mso.py [hunk 2]
```diff
@@ -853,6 +859,13 @@ def lookup_roles(self, roles, ignore_not_found_error=False):
             ids.append(dict(roleId=r.get("id"), accessType=access_type))
         return ids
 
+    def lookup_site_type(self, s):
```
Comment (akinross) [inline plugins/module_utils/mso.py:862]: Can you rename "s" to a more descriptive value. Found it confusing to understand where this platform key is now coming from. This is also due to the fact that in the module class we also have a platform defined with values "local", "nd" and "mso".

File: plugins/modules/mso_schema_site_service_graph.py [hunk 3]
```diff
@@ -225,29 +280,77 @@ def main():
...
-                        serviceNodeRef=dict(
-                            serviceNodeName=template_node_type,
-                            serviceGraphName=service_graph,
-                            templateName=template,
-                            schemaId=schema_id,
-                        ),
-                    ),
-                )
+                device_payload = dict()
+                if query_device_data:
```
Comment (akinross) [inline plugins/modules/mso_schema_site_service_graph.py:286]: So when query_device_data is None and empty dict will still be added to the payload? Is there a reason for this?

File: plugins/modules/mso_schema_site_service_graph.py [hunk 4]
```diff
@@ -225,29 +280,77 @@ def main():
...
+                    for device_data in query_device_data:
+                        if device.get("device_name") == device_data.get("dn").split("/")[-1].split("-")[-1]:
```
Comment (akinross) [inline plugins/modules/mso_schema_site_service_graph.py:288]: Can `device_data.get("dn").split("/")[-1].split("-")[-1]` ever lead to an error?

Comment (sajagana) [reply plugins/modules/mso_schema_site_service_graph.py:288]: The DN value points to the Cloud Site L4-L7 device, so it should follow the same pattern. Also, the DN format is common for the current L4-L7 devices.

Comment (akinross) [reply plugins/modules/mso_schema_site_service_graph.py:288]: ok

**Review by akinross — APPROVED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**